### PR TITLE
added history command, handled exceptions, refactoring

### PR DESCRIPTION
**Pull request by randrii**

### CODE REVIEW

File: src/main/java/com/rybka/command/HistoryCommand.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public void execute() {
+        try {
+            var histories = history();
+            print(histories);
+        } catch (Exception exception) {
+            log.error("Some issues have been occurred while returning history. Reason: " + exception.getMessage());
+        }
+    }
+
+    public List<CurrencyHistory> history() {
```
Comment (jarc0der) [inline src/main/java/com/rybka/command/HistoryCommand.java]: I think you don't need this method. Put the code into the `execute` method.

File: src/main/java/com/rybka/dao/CurrencyHistoryDAO.java [hunk 2]
```diff
@@ -28,7 +29,20 @@ public void save(CurrencyHistory currencyHistory) {
...
-            return session.createQuery("from CurrencyHistory", CurrencyHistory.class).list();
+            return session.createQuery(QueryConstants.SELECT_QUERY, CurrencyHistory.class).list();
+        } catch (Exception e) {
+            transaction.rollback();
+            log.error("Error while retrieving data from DB. Reason: " + e.getMessage());
+            throw new DBConnectionException("Unable to access to DB. Reason: " + e.getMessage());
+        }
+    }
+
+    public List<CurrencyHistory> findFiveRecords() {
```
Comment (jarc0der) [inline src/main/java/com/rybka/dao/CurrencyHistoryDAO.java]: findAllLimit(int count) would be better and reusable.

File: src/main/java/com/rybka/view/ExchangeView.java [hunk 3]
```diff
@@ -9,9 +9,11 @@
 public class ExchangeView {
 
     private final Command convertCommand;
+    private final Command historyCommand;
 
     public void showView() {
         convertCommand.execute();
+        historyCommand.execute();
```
Comment (jarc0der) [inline src/main/java/com/rybka/view/ExchangeView.java]: do we need to show history or convert values on showView? I believe we don't, till user provide appropriate command to execute.

File: src/main/java/com/rybka/command/HistoryCommand.java (hunk 1 above)
Comment (randrii) [reply src/main/java/com/rybka/command/HistoryCommand.java]: fixed

File: src/main/java/com/rybka/dao/CurrencyHistoryDAO.java (hunk 2 above)
Comment (randrii) [reply src/main/java/com/rybka/dao/CurrencyHistoryDAO.java]: fixed

File: src/main/java/com/rybka/view/ExchangeView.java (hunk 3 above)
Comment (randrii) [reply src/main/java/com/rybka/view/ExchangeView.java]: fixed

File: src/main/java/com/rybka/config/QueryConstants.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.rybka.config;
+
+public final class QueryConstants {
+    public static final String SELECT_QUERY = "FROM CurrencyHistory";
```
Comment (jarc0der) [inline src/main/java/com/rybka/config/QueryConstants.java:4]: CURRENCY_HISTORY_QUERY

**Pull request merged**